src/shared_modules/utils/tests/rocksDBWrapper_test.cpp

```diff
@@ -0,0 +1,278 @@
+    EXPECT_NO_THROW(db_wrapper->deleteAll());
+    std::string value {};
+    EXPECT_FALSE(db_wrapper->get("key6", value)); // The key should have been deleted
+    EXPECT_FALSE(db_wrapper->get("key7", value)); // The key should have been deleted
+}
+
+/**
+ * @brief Tests the deleteAll function with an empty database
+ */
+TEST_F(RocksDBWrapperTest, TestDeleteAllEmptyDB)
+{
+    Utils::RocksDBWrapper new_db_wrapper("new_test.db");
+    EXPECT_NO_THROW(db_wrapper->deleteAll());
+}
+
+/**
+ * @brief Tests the getIterator function
+ */
+TEST_F(RocksDBWrapperTest, TestIterator)
+{
+    const std::array<std::pair<std::string, std::string>, 4> elements {std::make_pair("key1", "value1"),
+                                                                       std::make_pair("key2", "value2"),
+                                                                       std::make_pair("key3", "value3"),
+                                                                       std::make_pair("key4", "value4")};
+
+    for (const auto& [key, value] : elements)
+    {
+        db_wrapper->put(key, value);
+    }
+
+    auto it {db_wrapper->getIterator()};
+
+    // Iterate through all the elements starting from the first one
+    auto counter {0};
+    for (it->SeekToFirst(); it->Valid(); it->Next())
+    {
+        EXPECT_EQ(it->key().ToString(), elements[counter].first);
+        EXPECT_EQ(it->value().ToString(), elements[counter].second);
+        ++counter;
+    }
+
+    // Start to iterate from the second element
+    counter = 2;
+    for (it->Seek(elements[counter].first); it->Valid(); it->Next())
+    {
+        EXPECT_EQ(it->key().ToString(), elements[counter].first);
+        EXPECT_EQ(it->value().ToString(), elements[counter].second);
+        ++counter;
+    }
+}
+
+/**
+ * @brief Tests the range for loop
+ */
+TEST_F(RocksDBWrapperTest, TestRangeForLoop)
+{
+    constexpr auto NUM_ELEMENTS {4};
+    constexpr auto NUM_ELEMENTS_ONE_MATCH {1};
+    const std::array<std::pair<std::string, std::string>, NUM_ELEMENTS> elements {std::make_pair("key1", "value1"),
+                                                                                  std::make_pair("key2", "value2"),
+                                                                                  std::make_pair("key3", "value3"),
+                                                                                  std::make_pair("key4", "value4")};
+    for (const auto& [key, value] : elements)
+    {
+        db_wrapper->put(key, value);
+    }
+
+    auto counter {0};
+
+    for (const auto& [key, value] : db_wrapper->seek("k"))
+    {
+        EXPECT_EQ(key, elements[counter].first);
+        EXPECT_EQ(value, elements[counter].second);
+        ++counter;
+    }
+
+    EXPECT_EQ(counter, NUM_ELEMENTS);
+
+    counter = 0;
+
+    for (const auto& [key, value] : db_wrapper->seek("key2"))
+    {
+        EXPECT_EQ(key, elements[counter + NUM_ELEMENTS_ONE_MATCH].first);
+        ++counter;
```

Comment:
This shouldn't be here.
This makes the test fail

<details><summary>Details</summary>
<p>

```bash
root@ubuntu-jammy:/home/vagrant/wazuh_2/wazuh/src/shared_modules/utils/tests/build# ./utils_unit_test 
[==========] Running 19 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 19 tests from RocksDBWrapperTest
[ RUN      ] RocksDBWrapperTest.TestPut
[       OK ] RocksDBWrapperTest.TestPut (26 ms)
[ RUN      ] RocksDBWrapperTest.TestPutEmptyKey
[       OK ] RocksDBWrapperTest.TestPutEmptyKey (12 ms)
[ RUN      ] RocksDBWrapperTest.TestPutEmptyValue
[       OK ] RocksDBWrapperTest.TestPutEmptyValue (13 ms)
[ RUN      ] RocksDBWrapperTest.TestPutExistingKey
[       OK ] RocksDBWrapperTest.TestPutExistingKey (10 ms)
[ RUN      ] RocksDBWrapperTest.TestPutExistingKeyUpdateValue
[       OK ] RocksDBWrapperTest.TestPutExistingKeyUpdateValue (12 ms)
[ RUN      ] RocksDBWrapperTest.TestGet
[       OK ] RocksDBWrapperTest.TestGet (37 ms)
[ RUN      ] RocksDBWrapperTest.TestGetNonExistentKey
Key not found: non_existent_key
[       OK ] RocksDBWrapperTest.TestGetNonExistentKey (18 ms)
[ RUN      ] RocksDBWrapperTest.TestGetEmptyKey
[       OK ] RocksDBWrapperTest.TestGetEmptyKey (15 ms)
[ RUN      ] RocksDBWrapperTest.TestGetEmptyDB
Key not found: key1
[       OK ] RocksDBWrapperTest.TestGetEmptyDB (37 ms)
[ RUN      ] RocksDBWrapperTest.TestDelete
Key not found: key3
[       OK ] RocksDBWrapperTest.TestDelete (14 ms)
[ RUN      ] RocksDBWrapperTest.TestDeleteNonExistentKey
[       OK ] RocksDBWrapperTest.TestDeleteNonExistentKey (15 ms)
[ RUN      ] RocksDBWrapperTest.TestDeleteEmptyKey
[       OK ] RocksDBWrapperTest.TestDeleteEmptyKey (11 ms)
[ RUN      ] RocksDBWrapperTest.TestDeleteEmptyDB
[       OK ] RocksDBWrapperTest.TestDeleteEmptyDB (22 ms)
[ RUN      ] RocksDBWrapperTest.TestGetLastKeyValue
[       OK ] RocksDBWrapperTest.TestGetLastKeyValue (13 ms)
[ RUN      ] RocksDBWrapperTest.TestGetLastKeyValueEmptyDB
[       OK ] RocksDBWrapperTest.TestGetLastKeyValueEmptyDB (23 ms)
[ RUN      ] RocksDBWrapperTest.TestDeleteAll
Key not found: key6
Key not found: key7
[       OK ] RocksDBWrapperTest.TestDeleteAll (15 ms)
[ RUN      ] RocksDBWrapperTest.TestDeleteAllEmptyDB
[       OK ] RocksDBWrapperTest.TestDeleteAllEmptyDB (55 ms)
[ RUN      ] RocksDBWrapperTest.TestIterator
[       OK ] RocksDBWrapperTest.TestIterator (17 ms)
[ RUN      ] RocksDBWrapperTest.TestRangeForLoop
/home/vagrant/wazuh_2/wazuh/src/shared_modules/utils/tests/rocksDBWrapper_test.cpp:253: Failure
Expected equality of these values:
  value
    Which is: "value2"
  elements[counter + NUM_ELEMENTS_ONE_MATCH].second
    Which is: "value3"
[  FAILED  ] RocksDBWrapperTest.TestRangeForLoop (20 ms)
[----------] 19 tests from RocksDBWrapperTest (412 ms total)

[----------] Global test environment tear-down
[==========] 19 tests from 1 test suite ran. (415 ms total)
[  PASSED  ] 18 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] RocksDBWrapperTest.TestRangeForLoop

 1 FAILED TEST
```

</p>
</details> 

How can we include these tests in Jenkins/Actions?